streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -689,6 +644,7 @@ void runOnce() {
              *  6. Otherwise, increment N.
              */
             do {
+                log.debug("Invoking TaskManager#process with {} iterations.", numIterations);
```

Comment:
What's the rationale of recording both the starting and the ending of a procedure? If it is for trouble shooting purposes only maybe the starting log entry can be trace while ending entry is debug?

Reply:
Yes, it's for debugging purposes, so I'd like to keep them both at debug level. When I was handling the incident that inspired this PR, what I really wanted to see in the logs is both how often we called poll and which operation in the loop was blocking. In theory, I'd only really need the "before" or the "after" logs for this, but only if I also have the source code pulled up to compare with the log messages. I'm proposing to (redundantly) log both "before" and "after" messages so that the logs will be context-free and people in the future would be able to tell what is happening just from the narrative of the logs themselves.

Reply:
SG